### PR TITLE
Fix system test generator option to be consistent

### PR DESCRIPTION
**Pull request by eileencodes**
We use `skip-system-test` everywhere else so it doesn't make sense to
use `system_tests = nil` in the generator config. Otherwise we're
passing conflicting options around (`system_tests` by default
`:test_unit` otherwise `nil` vs `skip_system_test` default is false and
otherwise pass true). It gets confusing.

Because of this change we then need to apply `skip_system_test` to the
`system_test` hook. In the PR to generate a more complete system test
when generating a scaffold this will be changed to be generated with
`create_test_files` but for now this works to unblock that PR.

Noticed this was incorrect while trying to figure out the test failures in https://github.com/rails/rails/pull/29418

### CODE REVIEW

**Comment by rafaelfranca** [comment]
I didn't understand what is the problem this PR is trying to solve. The changes you are doing here is reimplementing an already existing feature on thor (that is what was being used before this change).

Why do we have `system_tests` and `skip_system_test`? I think we only need `system_tests` because thor would already provide the skip version automatically.

**Comment by eileencodes** [comment]
Maybe I missed something obvious but how would thor pass the skip version automatically? In my testing the `--skip-system-test` option wasn't available until I added it.

System tests should be generated by default so it doesn't make sense to pass the following options to the generator: `bin/rails g scaffold --system-test=nil`.

Everywhere else we use `bin/rails g scaffold --skip-test` so it makes sense for this to be `bin/rails g scaffold --skip-system-test`.

Previously to this change the `system_test` config gets set to nil but the option never makes it to the `create_test_files` in #29418. I was going to add a `class_option :system_test` but that didn't make sense since everywhere else we're using `skip_system_test`.

If I leave the existing method then we'd need to have a `class_option :system_test` (for the generator config option) and a `class_option :skip_system_test` (for the command line).

**Comment by rafaelfranca** [comment]
I think we are mixing two things. One are generator hooks like test_unit, srylesheets, assets, etc and command line options like skip-bundler. What is currently implemented by the code is the hook. What your code is doing right is reimplementing the hook using a command line option.

The command line option skip-system-test can disable the system test hook like it is being done today, but the generator don't accept that command line option, only the application generator supports it. Are we planning to add that command line option to the generators?

If --skip-test is implemented as a command line option I'm fine with implementing the skip-system-test. Is skip test implemented in the same way?

**Comment by eileencodes** [comment]
Honestly I don't know. The generator code is super hard to follow. I'm trying to help Dinah in #29418 to figure out why the options don't come through at all here https://github.com/rails/rails/pull/29418/files#diff-2f84ab63853f4ddc12584cf05e56019bR22

This part of code is not aware of `options[:skip_system_test]` or of the option in the config file. What's the right way to get those options to that part of the generator?

As far as I can tell adding a `class_option` is the only way. When I added `skip_system_test` it wasn't picking up the generator config. Because that PR removes the `hook_for :system_test` because we no longer want to generate the default, we want to generate the one with create/update/delete from the controller actions when generating the scaffold. 

I don't think it makes sense to use both `skip_system_test` and `system_test` options. Here's Dinah's PR + my changes which I think will better demonstrate what I'm seeing.

File: railties/lib/rails/generators/rails/scaffold/scaffold_generator.rb [hunk 1]
```diff
@@ -12,16 +12,20 @@ class ScaffoldGenerator < ResourceGenerator # :nodoc:
...
-        @options = @options.merge(system_tests: false) if options[:api]
+        @options = @options.merge(skip_system_test: true) if options[:api]
       end
 
       hook_for :scaffold_controller, required: true
 
-      hook_for :system_tests, as: :system
+      hook_for :system_tests, as: :system do |system_test|
+        invoke system_test unless options[:skip_system_test]
+      end
```
Comment (eileencodes) [inline railties/lib/rails/generators/rails/scaffold/scaffold_generator.rb]: I think the change here is what you meant when you said "thor can skip without the extra configuration".

That's true and this is working on master. The problem comes in in #29418 because it no longer uses the `hook_for` since that would load the default plain system test template and we want to load the scaffold system test template with create/update/delete actions just like the controller test file does. In that case we need to pass the `class_option` to the scaffold generator. This was where I noticed that the config file was using `system_tests` as the `option` and not `skip_system_test`.

So I can either leave the config as is and pass `system_test` as a class_option to the scaffold generator, or change it as I did here.

**Comment by dinahshi** [comment]
From the discussion here, I was able to figure out how to get the `system_tests` option passed in. I've updated #29418 to reflect this.

I tried passing --skip-test as a command line option to generate scaffold on master and it doesn't look like this is currently supported. I'm not very familiar with thor but I think this implies that it's not necessary to change the option name.

**Comment by eileencodes** [comment]
Closing since this is no longer needed.
